src/core/ViewCubeGizmo.tsx

```diff
@@ -0,0 +1,126 @@
+import * as React from 'react'
+import { CanvasTexture, Event, Vector3 } from 'three'
+import { useGizmoHelper } from './GizmoHelper'
+
+const faces = ['RIGHT', 'LEFT', 'TOP', 'BOTTOM', 'FRONT', 'BACK']
+
+const edges: Array<[number, number, number]> = [
+  [1, 1, 0],
+  [1, 0, 1],
+  [1, 0, -1],
+  [1, -1, 0],
+  [0, 1, 1],
+  [0, 1, -1],
+  [0, -1, 1],
+  [0, -1, -1],
+  [-1, 1, 0],
+  [-1, 0, 1],
+  [-1, 0, -1],
+  [-1, -1, 0],
+]
+
+const corners: Array<[number, number, number]> = [
+  [1, 1, 1],
+  [1, 1, -1],
+  [1, -1, 1],
+  [1, -1, -1],
+  [-1, 1, 1],
+  [-1, 1, -1],
+  [-1, -1, 1],
+  [-1, -1, -1],
+]
+
+const FaceMaterial = ({ hover, index }: { hover: boolean; index: number }) => {
+  const texture = React.useMemo(() => {
+    const canvas = document.createElement('canvas')
```

Comment:
This might not work in `react-native` as you're using `document.createElement`? So this comp might need to go in `web` but tbh it's been a while since I've used `rn`.

Reply:
I noticed that `Effects.tsx` and `Shadow.tsx` use `document.createElement, so those probably have to move to `web` as well. 

To make this work for 'react-native' I could use `<Text/>` for the axis and face labels (like in https://codesandbox.io/s/r3f-render-target-kdj94?file=/src/index.js) but it seems heavy-handed for these simple labels. I could also check in textures instead of generating them dynamically, but then it's harder to localize this. Thoughts?

Reply:
they do, you're right, my bad. I wonder why they've never caused any issues.... okay let's leave as is 👌🏼 if people start having issues, we can look then.